third-party/realdds/include/realdds/dds-device-impl.h

```diff
@@ -206,117 +237,108 @@ class dds_device::impl
                             }
                             else
-                                LOG_ERROR( "Wrong message received, got 'SENSOR_HEADER' when the state is: " << state );
-
+                                LOG_ERROR( "Wrong message received, got 'DEVICE_HEADER' when the state is: " << state );
                             break;
 
                         case topics::device::notification::msg_type::VIDEO_STREAM_PROFILES:
                             if( state_type::WAIT_FOR_PROFILES == state )
                             {
-                                auto video_stream_profiles
-                                    = data.get< topics::device::notification::video_stream_profiles_msg >();
-
                                 LOG_INFO( "got VIDEO_STREAM_PROFILES message" );
 
-                                // TODO: Find a way to remove the "emplace" profiles copy
-                                topics::device::notification::video_stream_profiles_msg msg;
-                                msg.dds_sensor_index = video_stream_profiles->dds_sensor_index;
-                                msg.num_of_profiles = video_stream_profiles->num_of_profiles;
-                                for( int i = 0; i < video_stream_profiles->num_of_profiles; ++i )
-                                    msg.profiles[i] = video_stream_profiles->profiles[i];
+                                auto profiles_msg = data.get< topics::device::notification::video_stream_profiles_msg >();
+                                if ( profiles_msg == nullptr )
+                                {
+                                    DDS_THROW( runtime_error, "get< topics::device::notification::video_stream_profiles_msg > return nullptr" );
+                                }
 
-                                auto sensor_name_it
-                                    = _sensor_index_to_name.find( video_stream_profiles->dds_sensor_index );
-                                if( sensor_name_it != _sensor_index_to_name.end() )
+                                for ( size_t i = 0; i < profiles_msg->num_of_profiles; ++i )
                                 {
-                                    _sensor_to_video_profiles.emplace( video_stream_profiles->dds_sensor_index,
-                                                                       std::move( msg ) );
-
-                                    if( _sensor_to_video_profiles.size() + _sensor_to_motion_profiles.size()
-                                        < _num_of_sensors )
-                                        state = state_type::WAIT_FOR_SENSOR_HEADER;
-                                    else
-                                        state = state_type::DONE;
+                                    auto profile = profiles_msg->profiles[i];
+                                    auto key = std::make_pair( profile.uid, profile.stream_index );
+                                    auto & stream = _streams[key];
```

Comment:
If I am not mistaking this line will always generate a new entry ,
I am not sure the following `!` condition will ever be met.
Am I mistaking and this syntax acts differently?

Reply:
Will generate an entry if does not exist.
`_streams` maps `shared_ptr`s, the appropriate `!` operator will be called.

See previous discussion opened by @maloel, suggesting this syntax.